qiskit/synthesis/multi_controlled/mcx_synthesis.py

```diff
@@ -59,7 +67,8 @@ def synth_mcx_n_dirty_i15(
         qc.ccx(q_controls[0], q_controls[1], q_target)
         return qc
     elif not relative_phase and num_ctrl_qubits == 3:
-        qc._append(C3XGate(), [*q_controls, q_target], [])
+        circuit = synth_c3x()
+        qc.append(circuit, [*q_controls, q_target])
```

Comment:
I'm wondering whether we should skip the extra wrapping here, which would be more performant in unrolling and save us circuit copies in the construction.
```suggestion
        qc.compose(synth_c3x(), [*q_controls, q_target], inplace=True, copy=False)
```

Reply:
I actually tried this, but there is a case when the circuit has more qubits than the decomposition provided by `synth_c3x()`. In this case, `append` works correctly, but `compose` (wrongly) does not do anything. But I probably need to specify the qubit mapping correctly.

Reply:
Yeah `compose` should work 🤔 but not doing anything is strange, that would mean that either the circuit was empty or maybe `inplace=True` was missing?

Reply:
Ah, `inplace=True` solved it.